modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheProxy.java

```diff
@@ -780,6 +780,13 @@ private ClusterGroup projection(boolean loc) {
 
                 final SqlQuery p = (SqlQuery)qry;
 
+                if (ctx.isReplicated() && p.getPartitions() != null)
```

Comment:
Added eager check for isReplicatedOnly mode and another when all caches are prepared and determined to be replicated only. In such case setting explicit partitions will throw an exception.